### DISCUSSION TITLE
SSDP - making TAS appear in Windows as a clickable 'Other Device'

### DISCUSSION THREAD

**Discussion by btsimonh** [arendst/Tasmota#17084] [Ideas]
I've not got an active development Tas device at the moment, but I have been playing with some other projects.

In one, I implemented SSDP so that the device appeared in Windows networking as an 'Other Device' which when clicked opened the browser to it.

In TAS we have Wemo emulation, which uses SSDP (also hue emulation, the same?).
If you enable Wemo emulation, then TAS appears as a device, but without click action.

It is probably that adding the line 
```
      "</serviceList>"
      "<presentationURL>http://%s:80/</presentationURL>"
    "</device>"
```

in tasmota\tasmota_xdrv_driver\xdrv_21_wemo_multi.ino would enable clicking on the TAS device in windows to launch the browser against that device.
(and here tasmota\tasmota_xdrv_driver\xdrv_21_wemo.ino ?)
(and possibly here tasmota\tasmota_xdrv_driver\xdrv_20_hue.ino ?)

What I don't know is if this will disturb anything looking for a real wemo.....

This might be a useful addition...
thoughts?

**Comment by barbudor** [comment]
That's a nice finding
Not sure I would use it myself but it could be interresting for beginners with limited number of devices

**Comment by btsimonh** [comment]
I missed it when I screwed up the firmware and it did not work any more :).
But it's definately something that should be configurable (maybe on by default), as some don;t line the UPNP traffic.
It should be noted that Tas already shows up, it's just that when you double click, you then have to copy/paste the IP to the browser... (only if wemo emulation enabled).
What is a little strange in TAS is that Wemo emulation runs a second webserver - that must be costly in resource when it could use the existing one like Philips hue emulation.....

**Reply by arendst** [reply]
The second webserver was added recently to allow WEMO emulation as multilple devices which WEMO, by design, does not support.

**Reply by btsimonh** [reply]
:)  I knew there would be a good reason.  So we appear as multiple devices....

**Comment by btsimonh** [comment]
Also, TAS does not send (multicast) NOTIFY.
In OBK, I do this, and catch my NOTIFY in other devices to keep a simple device list - so from the webapp, we can navigate to any other OBK device from the list....

**Comment by arendst** [comment]
I prefer not to tinker with the WEMO and/or Hue implementation but add a new (first optional) one allowing tweaking it to it's minimal size and functionality without disturbing what we already have.

**Comment by arendst** [comment]
Just implemented this in the dev branch on Wemo emulation only

To use it define USE_UNISHOX_COMPRESSION needs to be disabled (as the feature is not yet in the compressed XML).

Works on my windows 11 box!

**Reply by btsimonh** [reply]
I thought you may like it :).
When you look at the separate driver, the NOTIFY aspect, and 'hearing' other notify from other TAS is a sweet way to join them all together.
